src/api/tests/service/shifu/test_admin_users.py

```
@@ -2967,6 +2970,123 @@ def test_grant_operator_user_package_is_idempotent_for_repeated_request_id(app):
+        subscription = (
+            BillingSubscription.query.filter_by(
+                creator_bid="package-grant-pingxx-upgrade",
+                subscription_bid=result.subscription_bid,
+            )
+            .order_by(BillingSubscription.id.desc())
+            .first()
+        )
+
+    assert result.product_bid == "bill-product-plan-yearly"
+    assert order is not None
+    assert order.order_type == BILLING_ORDER_TYPE_SUBSCRIPTION_UPGRADE
```

Ruff (0.15.14)

[error] Undefined name `BILLING_ORDER_TYPE_SUBSCRIPTION_UPGRADE`

(F821)